LoopCore/LoopSettings.swift

```diff
@@ -298,3 +298,16 @@ extension LoopSettings: RawRepresentable {
         return raw
     }
 }
+
+extension LoopSettings {
+    public var therapySettings: TherapySettings {
+        TherapySettings(glucoseTargetRangeSchedule: self.glucoseTargetRangeSchedule,
+                        preMealTargetRange: self.preMealTargetRange,
+                        workoutTargetRange: self.legacyWorkoutTargetRange,
+                        maximumBasalRatePerHour: self.maximumBasalRatePerHour,
+                        maximumBolus: self.maximumBolus,
+                        suspendThreshold: self.suspendThreshold,
+                        insulinSensitivitySchedule: nil, // TODO
+                        carbRatioSchedule: nil) // TODO
```

Comment:
As you're finding, in the current architecture TherapySettings doesn't compose nicely into LoopSettings.

I'd recommend one of the following:
- Turn the `therapySettings` computed property into a function that takes these schedules as arguments (I suspect you'll need the basal schedule too)
- Make `therapySettings` a computed property on LoopDataManager, where all of the required data is available.